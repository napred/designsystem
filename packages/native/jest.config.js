const { resolve } = require('path');

module.exports = {
  collectCoverageFrom: ['src/**/*.{js,jsx,ts,tsx}'],
  displayName: '@napred/native',
  globals: {
    'ts-jest': {
      diagnostics: true,
      tsConfig: resolve(__dirname, './tsconfig.json'),
    },
  },
  haste: {
    defaultPlatform: 'ios',
    platforms: ['android', 'ios', 'native'],
    hasteImplModulePath: '<rootDir>/node_modules/react-native/jest/hasteImpl.js',
    providesModuleNodeModules: ['react-native'],
  },
  rootDir: resolve(__dirname, './../../'),
  modulePathIgnorePatterns: [
    resolve(__dirname, './../../node_modules/react-native/Libraries/react-native/'),
  ],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'json'],
  moduleNameMapper: {
    '@napred/ds': resolve(__dirname, '../ds/src'),
    '^react$': resolve(__dirname, '../../node_modules/react'),
    '^React$': resolve(__dirname, '../../node_modules/react'),
  },
  testMatch: ['<rootDir>/packages/native/src/**/__tests__/*.(test|spec).{js,jsx,ts,tsx}'],
  transform: {
    '^.+\\.(bmp|gif|jpg|jpeg|mp4|png|psd|svg|webp)$':
      '<rootDir>/node_modules/react-native/jest/assetFileTransformer.js',
    '\\.tsx?$': 'ts-jest',
    '\\.js$': '<rootDir>/node_modules/react-native/jest/preprocessor.js',
  },
  transformIgnorePatterns: ['node_modules/(?!(jest-)?react-native|react-clone-referenced-element)'],
  setupFiles: ['<rootDir>/node_modules/react-native/jest/setup.js'],
  testEnvironment: 'node',
};
