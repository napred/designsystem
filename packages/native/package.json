{
  "name": "@napred/native",
  "version": "0.7.0",
  "main": "dist/native.cjs.js",
  "jsnext:main": "dist/native.m.js",
  "module": "dist/native.m.js",
  "typings": "dist/index.d.js",
  "keywords": [
    "react",
    "design-system",
    "css-in-js",
    "ui",
    "react-native"
  ],
  "contributors": [
    {
      "name": "Michal Kvasničák",
      "url": "https://github.com/michalkvasnicak"
    },
    {
      "name": "Juraj Hríb",
      "url": "https://github.com/jurajhrib"
    }
  ],
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "peerDependencies": {
    "@types/react": "^16.7.13",
    "@types/react-native": "^0.57.17",
    "react": "^16.7.0-alpha.2 || >=16.7.0",
    "react-native": "^0.57.7"
  },
  "dependencies": {
    "@napred/ds": "^0.7.0",
    "css-mediaquery": "^0.1.2",
    "css-to-react-native": "^2.2.2",
    "stylis": "^3.5.4"
  },
  "devDependencies": {
    "react-native-testing-library": "^1.4.2",
    "typescript": "^3.2.2"
  },
  "gitHead": "3d6bf1ca4790f3c79c2b1e8dc2d7ed03a2388319",
  "scripts": {
    "build": "tsc && rollup -c",
    "lint": "tslint -p .",
    "prebuild": "rimraf dist temp",
    "prepublishOnly": "npm run build",
    "test": "jest",
    "test:watch": "jest --watch",
    "typecheck": "tsc --noEmit"
  },
  "sideEffects": false
}
