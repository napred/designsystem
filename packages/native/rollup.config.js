import nodeResolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import babel from 'rollup-plugin-babel';

const input = 'temp/index.js';
const external = id => !id.startsWith('\0') && !id.startsWith('.') && !id.startsWith('/');

const babelCJS = {
  babelrc: false,
  presets: [['@babel/preset-env', { modules: false }], '@babel/preset-react'],
  exclude: /node_modules/,
};

const babelESM = {
  babelrc: false,
  exclude: /node_modules/,
  runtimeHelpers: true,
  presets: [['@babel/preset-env', { modules: false }], '@babel/preset-react'],
};

export default [
  // cjs
  {
    external,
    input,
    output: {
      exports: 'named',
      file: 'dist/native.cjs.js',
      format: 'cjs',
    },
    plugins: [
      nodeResolve({
        extensions: ['.mjs', '.js', '.jsx'],
      }),
      commonjs({
        ignoreGlobal: true,
      }),
      babel(babelCJS),
    ],
  },

  // esm
  {
    input,
    external,
    output: {
      file: 'dist/native.m.js',
      format: 'esm',
    },
    plugins: [
      babel(babelESM),
      nodeResolve({
        extensions: ['.mjs', '.js', '.jsx'],
      }),
      commonjs({
        ignoreGlobal: true,
      }),
    ],
  },
];
