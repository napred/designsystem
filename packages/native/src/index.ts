export * from './createStyleApplicator';
export * from './DesignSystem';
export * from './types';

export { default as createComponent } from './createComponent';
export { default as createStyle } from './createStyle';
export { default as defaultTheme } from './defaultTheme';
export { default as DesignSystem } from './DesignSystem';

export * from '@napred/ds';
