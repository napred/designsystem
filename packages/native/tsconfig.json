{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "composite": true,
    "declarationDir": "dist",
    "lib": ["es2018"],
    "outDir": "temp",
    "rootDir": "src",
    "target": "esnext",
    "typeRoots": ["node_modules/@types", "@types", "../../node_modules/@types"]
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["dist", "node_modules", "src/**/__tests__/**/*", "src/**/__mocks__/**/*"],
  "references": [{ "path": "../ds" }]
}
