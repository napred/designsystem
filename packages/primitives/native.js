module.exports = require('./dist/primitives-native.cjs');