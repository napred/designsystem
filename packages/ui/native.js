module.exports = require('./dist/ui-native.cjs');
