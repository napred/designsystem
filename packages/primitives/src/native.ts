import { createComponent } from '@napred/native';
import { Image as NativeImage, Text as NativeText, View } from 'react-native';
import * as BoxDef from './Box';
import * as CardDef from './Card';
import * as FlexDef from './Flex';
import * as ImageDef from './Image';
import * as LinkDef from './Link';
import * as TextDef from './Text';
import * as TitleDef from './Title';

export const Box = createComponent(BoxDef.name, View);

Box.defaultProps = BoxDef.defaultProps;

export const Card = createComponent(CardDef.Card.name, View);
export const CardFooter = createComponent(CardDef.CardFooter.name, View);
export const CardHeader = createComponent(CardDef.CardHeader.name, View);

Card.defaultProps = CardDef.Card.defaultProps;
CardFooter.defaultProps = CardDef.CardFooter.defaultProps;
CardHeader.defaultProps = CardDef.CardHeader.defaultProps;

export const Flex = createComponent(FlexDef.name, View);

Flex.defaultProps = FlexDef.defaultProps;

export const Image = createComponent(ImageDef.name, NativeImage);

// @ts-ignore - for now, because we don't have a check for string components, so it should error in browser too
Image.defaultProps = ImageDef.defaultProps;

export const Link = createComponent(LinkDef.name, NativeText, {
  style: LinkDef.style,
});

Link.defaultProps = LinkDef.defaultProps;

export const Text = createComponent(TextDef.name, NativeText);

Text.defaultProps = TextDef.defaultProps;

export const Title = createComponent(TitleDef.name, NativeText);

Title.defaultProps = TitleDef.defaultProps;
