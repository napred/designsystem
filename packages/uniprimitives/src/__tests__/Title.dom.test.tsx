/**
 * @jest-environment jsdom
 */
import { DesignSystem } from '@napred/native';
import React from 'react';
import { render } from 'react-testing-library';
import { Title } from '../';

describe('Title', () => {
  it('renders as anchor in DOM', () => {
    const { asFragment } = render(
      <DesignSystem>
        <Title>Title</Title>
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
