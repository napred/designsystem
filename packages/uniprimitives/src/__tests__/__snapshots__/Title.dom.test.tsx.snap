// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders as anchor in DOM 1`] = `
<DocumentFragment>
  <h1
    aria-level="1"
    dir="auto"
    role="heading"
    style="border-top-width: 0px; border-right-width: 0px; border-bottom-width: 0px; border-left-width: 0px; box-sizing: border-box; display: inline; font-family: Palanquin; font-size: 16px; font-style: inherit; font-variant: inherit; font-weight: bold; line-height: 20.8px; margin: 0px 0px 0px 0px; padding: 0px 0px 0px 0px; text-decoration: none; white-space: pre-wrap; word-wrap: break-word;"
  >
    Title
  </h1>
</DocumentFragment>
`;
